### PR TITLE
Size package

### PR DESCRIPTION
**Pull request by kujenga**
This PR adds a `size` package useful for determining the size of objects. This initial PR introduces support for the size of minified JSON objects.

### CODE REVIEW

**Comment by kujenga** [comment]
The mail chimp cleanup in here was because I had to fix an unrelated issue with the Mailchimp list that the tests were running against, and saw the opportunity to to a little maintenance.

File: size/json.go [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+		if err != nil {
+			debugln("size: error marshaling unhandled type: ", err)
+		}
+		return len(data)
+	}
+}
+
+func timeLen(t *time.Time) int {
+	// the default time format for marshalled time
+	base := len(time.RFC3339Nano)
```
Comment (AndyHaskell) [inline size/json.go]: I think it might make sense to have a string param to `timeLen`/`JSON` for other timestamp formats

File: size/json.go [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+	_, offset := t.Zone()
+	if offset == 0 {
+		// handle "Z" UTC indicator
+		base -= len("00:00")
```
Comment (AndyHaskell) [inline size/json.go]: Should this be `len("Z00:00")`?

File: size/json.go (hunk 1 above)
Comment (kujenga) [inline size/json.go]: What situation are you referring to? JSON marshaling for `time.Time` is always in RFC3339 I believe: https://golang.org/src/time/time.go?s=26891:26934#L918

File: size/json.go (hunk 2 above)
Comment (kujenga) [inline size/json.go]: No, there is never both a Z and an explicit time offset. Having both is invalid, there's more info in the `time` package docs: https://golang.org/pkg/time/

File: size/json.go (hunk 1 above)
Comment (kujenga) [inline size/json.go]: If you're referring to custom types that alias `time.Time`, those won't be handled by this logic, instead falling through to the default case.

**Comment by AndyHaskell** [comment]
:+1:

**Pull request merged**